### ISSUE TITLE
Improved Denoising Diffusion Probabilistic Models

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#155]
## TL; DR

- Moderate technical modifications empower DDPMs with log-likelihood comparable to SOTA methods and save Markov transition steps for sampling.
- Evaluate DDPMs with additional metric (Precision/Recall) to demonstrate its strength in mode preservation.
- Examine the scalability of DDPM with larger and more complex dataset.

## Motivation

- DDPMs have shown to achieve SOTA performance in sample quality on relatively simple dataset, but their likelihood estimation fell short.
- Thorough evaluation on more complex dataset or sample diversity was missing in the original paper, as well as empirical results from scale-up experiments.

## Prior Works

- DDPM

## Method

- Learn forward variance with interpolation parameter (between two extreme choices)
- Incorporate cosine noise schedule (\beta)
- Suggest hybrid objective that uses both simplified version of (Ho et. al., 2020) and the original VLB.

## Experiments

![image](https://user-images.githubusercontent.com/46308407/152935848-0676310b-f7f7-405d-b5da-db32d9397f27.png)

- Except for really early steps, these two values are almost identical (hence not much difference in Ho et. al.'s experiments)

![image](https://user-images.githubusercontent.com/46308407/152935919-dd905e13-a816-4271-9592-4e0e027cae16.png)

- Cosine noise schedules efficiently saves tedious denoising steps and refines the overall dynamics.
- With linear noise schedules, a significant fraction of reverse steps are effectively *redundant*.

![image](https://user-images.githubusercontent.com/46308407/152936158-c6388f8b-f07d-4d60-817f-95110a660793.png)

- Reducing gradient noise by importance sampling t (rather than uniform) helped stabilizing training.

![image](https://user-images.githubusercontent.com/46308407/152936280-316e6baf-2c19-4eb4-9fff-7172e71791de.png)
- However gain from their modifications is relatively small, in my opinion.
- Gain from using more denoising steps is also meaningful.

![image](https://user-images.githubusercontent.com/46308407/152936438-3e5e953d-3710-4693-a480-ae9406c20193.png)

- The authors did succeed in roughly matching the SOTA log-likelihood on popular benchmarks.

![image](https://user-images.githubusercontent.com/46308407/152936567-deebcd06-b802-4534-bc97-1bf6ffbb4f83.png)

- Despite being much slower to sample, metrics are indeed favorable to DDPMs even when compared to SOTA GAN model like BigGAN-deep.
- Especially 'recall' metric is very good, indicating better mode preservation property of DDPMs. (We should note that minimal truncation was used for BigGAN-deep to ensure best recall)

![image](https://user-images.githubusercontent.com/46308407/152936627-a18e8540-7849-46ea-8a45-6f324805d0b4.png)
- Expanding model width boosts performance in near-linear fashion.

## Comments

- Good review on DDPM (*Background section was really well-written*)
- Personally think the technical novelty is not that great, but the authors have provided us with valuable evaluations (scale-up, precision/recall, log-likelihood) that can be foundational to future works in the field.
